shell/browser/electron_browser_client.cc

```diff
@@ -455,6 +455,13 @@ void ElectronBrowserClient::AppendExtraCommandLineSwitches(
     base::CommandLine* command_line,
     int process_id) {
   // Make sure we're about to launch a known executable
+#if BUILDFLAG(IS_LINUX)
+  // On Linux, do not perform this check for /proc/self/exe. It will always
+  // point to the currently running executable so this check is not
+  // necessary, and if the executable has been deleted it will return a fake
+  // name that causes this check to fail.
+  if (command_line->GetProgram() != base::FilePath("/proc/self/exe"))
```

Comment:
This current check feels a bit brittle to me - Chromium does have some handling for this though and I think we can use that approach instead.

First - Chrome exports [`kProcSelfExe`](https://source.chromium.org/chromium/chromium/src/+/main:base/process/process_metrics.h;drc=3cc22c22d7637bc5604e8fef3b0882b51a762901;bpv=1;bpt=1;l=416) and we should use that instead of a hardcoded string here.

Secondly - handling for this specific case is already built into `base::PathService::Get` [here](https://source.chromium.org/chromium/chromium/src/+/main:base/base_paths_posix.cc;l=44-47;drc=3cc22c22d7637bc5604e8fef3b0882b51a762901;bpv=1;bpt=1). I think that instead of adding this wrapper `if` statement here, we should instead check to see if `command_line->GetProgram()` is `/proc/self/exe` in the event that `base::PathService::Get` fails, and only CHECK if is _not_.

Reply:
> First - Chrome exports [`kProcSelfExe`](https://source.chromium.org/chromium/chromium/src/+/main:base/process/process_metrics.h;drc=3cc22c22d7637bc5604e8fef3b0882b51a762901;bpv=1;bpt=1;l=416) and we should use that instead of a hardcoded string here.

Sounds good.

> Secondly - handling for this specific case is already built into `base::PathService::Get` [here](https://source.chromium.org/chromium/chromium/src/+/main:base/base_paths_posix.cc;l=44-47;drc=3cc22c22d7637bc5604e8fef3b0882b51a762901;bpv=1;bpt=1). I think that instead of adding this wrapper `if` statement here, we should instead check to see if `command_line->GetProgram()` is `/proc/self/exe` in the event that `base::PathService::Get` fails, and only CHECK if is _not_.

Actually, `base::PathService::Get` succeeds. It looks like it's cached (and even if it wasn't, `readlink("/proc/self/exe")` still succeeds even after the executable is unlinked). However, `base::MakeAbsoluteFilePath` fails and returns an empty string, which causes the `CHECK_EQ` a few lines down to fail.

I could change this to skip that `CHECK_EQ` if `command_line->GetProgram()` is `/proc/self/exe`?